include/moqt/publish_track_handler.h

```diff
@@ -0,0 +1,267 @@
+     *
+     *  This extends the base track handler to add publish (aka send) handling
+     */
+    class PublishTrackHandler : protected BaseTrackHandler
+    {
+      public:
+        friend class Transport;
+
+        /**
+         * @brief Publish status codes
+         */
+        enum class PublishStatus : uint8_t
+        {
+            kOk = 0,
+            kInternalError,
+            kNotAuthorized,
+            kNotAnnounced,
+            kNoSubscribers,
+            kObjectPayloadLengthExceeded,
+            kPreviousObjectTruncated,
+
+            kNoPreviousObject,
+            kObjectDataComplete,
+            kObjectContinuationDataNeeded,
+            kObjectDataIncomplete,              ///< PublishObject() was called when continuation data remains
+
+            /// Indicates that the published object data is too large based on the object header payload size plus
+            /// any/all data that was sent already.
+            kObjectDataTooLarge,
+
+            /// Previous object payload has not been completed and new object cannot start in per-group track mode
+            /// unless new group is used.
+            kPreviousObjectNotCompleteMustStartNewGroup,
+
+            /// Previous object payload has not been completed and new object cannot start in per-track track mode
+            /// without creating a new track. This requires to unpublish and to publish track again.
+            kPreviousObjectNotCompleteMustStartNewTrack,
+        };
+
+        /**
+         * @brief  Status codes for the publish track
+         */
+        enum class Status : uint8_t
+        {
+            kOK = 0,
+            kNotConnected,
+            kNotAnnounced,
+            kPendingAnnounceResponse,
+            kAnnounceNotAuthorized,
+            kNoSubscribers,
+            kSendingUnannounce,
+        };
+
+
+        // --------------------------------------------------------------------------
+        // Public API methods that normally should not be overridden
+        // --------------------------------------------------------------------------
+
+        /**
+         * @brief Publish track handler constructor
+         *
+         * @param full_track_name       Full track name struct
+         * @param track_mode            The track mode to operate using
+         * @param default_priority      Default priority for objects if not specified in ObjectHeaderss
+         * @param default_ttl           Default TTL for objects if not specified in ObjectHeaderss
+         */
+        PublishTrackHandler(const FullTrackName& full_track_name,
+                            TrackMode track_mode,
+                            uint8_t default_priority,
+                            uint32_t default_ttl)
+          : BaseTrackHandler(full_track_name)
+          , track_mode_(track_mode)
+          , def_priority_(default_priority)
+          , def_ttl_(default_ttl)
```

Comment:
s/def/default

Reply:
Implementation is not part of this review. This review is only for public API's. Implementation has not been updated or changed yet to implement the public API changes. Private members, such as the above are not part of the public API. These are internal use member variables that are for the implementation.

Will rename, but according to google style this is allowed in this scope.